### PR TITLE
Test metadata helper function consolidation

### PR DESCRIPTION
**Pull request by camsaul**
* `lib.tu/mock-cards` now has metadata in the same order as query results
* Add `no-metadata` versions to `lib.tu/mock-cards` e.g. `:venues/no-metadata`
* Add `native` versions to `lib.tu/mock-cards` e.g. `:venues/native`
* Remove `lib.tu/categories-mbql-card`, you can just use `lib.tu/mock-cards` instead
* Remove `lib.tu/metadata-provider-with-categories-mbql-card`, you can just use `lib.tu/mock-cards` instead
* Remove `lib.tu/categories-native-card`, you can just use `lib.tu/mock-cards` instead
* Remove `lib.tu/metadata-provider-with-categories-native-card`, you can just use `lib.tu/mock-cards` instead
* Remove `meta/card-results-metadata`, you can just use `lib.tu/mock-cards` instead
* Move `qp-results-metadata` to the one test namespace that it's actually used
* Remove `saved-question-CardMetadata`, you can just use `lib.tu/mock-cards` instead
* Remove `saved-question-CardMetadata`, you can just use `lib.tu/mock-cards` instead
* Move `lib.query/saved-question-query` to `lib.tu/query-with-stage-metadata-from-card`, since it's only used in tests; rename to better reflect what it actually does. Replace usages that don't set metadata with calls to `lib/query`, since it's effectively the same thing.
* `lib/query` now calls `lib.convert/->pMBQL` and thus handles legacy queries
* Add `lib.tu/query-with-mock-card-as-source-card` for easily creating a query with a `:source-card`

### CODE REVIEW

File: test/metabase/lib/column_group_test.cljc [hunk 1]
```diff
@@ -275,31 +276,31 @@
...
                       ::lib.column-group/group-type :group-type/join.explicit
-                      ::lib.column-group/columns    [{:name "ID", :lib/card-id 1}
-                                                     {:name "NAME", :lib/card-id 1}]}]
+                      ::lib.column-group/columns    [{:name "ID", :lib/card-id (:id card)}
+                                                     {:name "NAME", :lib/card-id (:id card)}]}]
                     groups)))
           (testing `lib/display-info
             (testing "Card is not present in MetadataProvider"
-              (is (=? [{:display-name "Saved Question 1"
+              (is (=? [{:display-name (lib.card/fallback-display-name (:id card))
```
Comment (snoe) [inline test/metabase/lib/column_group_test.cljc:296]: surprised to see this call in an expected test value.

File: test/metabase/lib/test_util.cljc [hunk 2]
```diff
@@ -223,69 +223,78 @@
...
-  "A metadata provider with the [[categories-native-card]] as Card 1. Composed with the
-  normal [[meta/metadata-provider]]."
-  (lib.metadata.composed-provider/composed-metadata-provider
-   meta/metadata-provider
-   (mock-metadata-provider
-    {:cards [categories-native-card]})))
-
 (def mock-cards
-  "Map of mock MBQL query Card against the test tables."
+  "Map of mock MBQL query Card against the test tables. There are two versions of the Card for each table:
```
Comment (snoe) [inline test/metabase/lib/test_util.cljc:227]: *three versions

**Review by snoe — APPROVED** [review]
This is great, thanks!

File: test/metabase/lib/column_group_test.cljc (hunk 1 above)
Comment (camsaul) [reply test/metabase/lib/column_group_test.cljc:296]: good point, I will change it back

File: test/metabase/lib/test_util.cljc (hunk 2 above)
Comment (camsaul) [reply test/metabase/lib/test_util.cljc]: fixed

**Pull request merged**